### PR TITLE
Feature/ontology hierarchy

### PR DESCRIPTION
**Pull request by pelmat**
# Feature/ontology hierarchy

## Support for full YSO ontology hierarchy, ontology types. Keyword search improvements. New ontology importer.
-----------------------------------------------------------------------------------------------
#### Changes
1. events/admin.py
     * Added support for keyword parent, children, ontology type editing.

2. events/api.py
     * Added support for keyword hierarchies, ontology types.
     * Keyword searches are now ordered by result accuracy and filtered by locale.
     * alt_labels are used only if search results comes out empty without them.
     * Added simple ordering to viewsets missing them for more consistent pagination results.

3. events/importer/ontology.py
     * New, improved ontology importer that supports keyword hierarchies, ontology types.

4. events/migrations/0084_auto_20220118_1006.py
     * Migration file for keyword model changes.
    
5. events/models.py
     * Added support for keyword hierarchies, ontology types.

6. events/utils.py
     * New util needed for "@ id" url generation.

### CODE REVIEW

File: events/importer/ontology.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+logFile = \
+    logging.FileHandler(
+        '%s' % (join(dirname(__file__), 'logs', curFile+'.logs'))
+    )
+logFile.setFormatter(
+    logging.Formatter(
+        '[%(asctime)s] <%(name)s> (%(lineno)d): %(message)s'
+    )
+)
+logFile.setLevel(logging.DEBUG)
```
Comment (aceViilee) [inline events/importer/ontology.py:44]: When making production release logging level change required

**Review by aceViilee — APPROVED** [review]

File: events/importer/ontology.py (hunk 1 above)
Comment (aceViilee) [reply events/importer/ontology.py:44]: Should consider making it env variable dependant

File: events/migrations/0084_auto_20220118_1006.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Generated by Django 2.2.24 on 2022-01-18 08:06
```
Comment (juuso-j) [inline events/migrations/0084_auto_20220118_1006.py]: Tip, rename migrations descriptive, e.g. 0084_created_fields_children_ontology_type_and_parent.py. Makes it easier when i.e. reverting migrations.

**Review by juuso-j — APPROVED** [review]

**Review by ezkat — APPROVED** [review]

File: events/migrations/0084_auto_20220118_1006.py (hunk 2 above)
Comment (pelmat) [reply events/migrations/0084_auto_20220118_1006.py]: Changed the name of the latest migration file to something more apt and also fixed a dependency issue that came with the previous merge from development-branch.

**Comment by ghost** [comment]
I'd like to add a small developer apology for my lack of understanding in git workflow.

I made a big mistake not committing the changes from the collaborative sessions I made with pelmat on the same branch on the same server. For example, the entire ontology importer and the search result and algorithm additions / fixes were done by me.

**Pull request merged**